bpython/curtsiesfrontend/repl.py

```diff
@@ -1137,11 +1130,26 @@ def clear_current_block(self, remove_from_history=True):
     def get_current_block(self):
         return '\n'.join(self.buffer + [self.current_line])
 
+    def move_current_stdouterr_line_up(self):
+        """Append self.current_stdouterr_line to self.display_lines
+        then clean it."""
+        self.display_lines.extend(paint.display_linize(
+            self.current_stdouterr_line, self.width))
+        self.current_stdouterr_line = ''
+
     def send_to_stdout(self, output):
         """Send unicode string to Repl stdout"""
+        if not output: return
         lines = output.split('\n')
+        if output == len(output) * '\n':
+            # If the string consist only of newline characters,
+            # str.split returns one more empty strings.
```

Comment:
Since at this point `split` was already called, I'd feel that `if all(len(line) == 0 for line in lines):` would be nicer.